spacy/pipeline/entity_linker.py

```diff
@@ -212,10 +235,23 @@ def initialize(
         doc_sample = []
         vector_sample = []
         for example in islice(get_examples(), 10):
-            doc_sample.append(example.x)
+            doc = example.x
+            if self.use_gold_ents:
+                doc.ents = example.y.ents
+            doc_sample.append(doc)
             vector_sample.append(self.model.ops.alloc1f(nO))
         assert len(doc_sample) > 0, Errors.E923.format(name=self.name)
         assert len(vector_sample) > 0, Errors.E923.format(name=self.name)
+
+        # XXX This is a hack. Also `has_annotation` doesn't seem to work as expected?
+        # Check that there are some entities in the batch
+        has_annotations = any([doc.ents for doc in doc_sample])
+        if not has_annotations:
+            doc = doc_sample[0]
+            ent = doc[0:1]
```

Comment:
What if this `doc` happens to be empty (ie no tokens)? Can you write a unit test for this edge case?

Reply:
I added a test and check for when the batch has no usable gold examples, which will also cover the no tokens case.